pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -663,6 +688,75 @@ func GetShortNameMode(ctx *types.SystemContext) (types.ShortNameMode, error) {
 	return config.shortNameMode, err
 }
 
+// GlobalCredentialHelpers returns the global top-level credential helpers.
+// A slice with only AuthenticationFileHelper is returned if no helpers are
+// configured.
+func GlobalCredentialHelpers(sys *types.SystemContext) ([]string, error) {
+	config, err := getConfig(sys)
+	if err != nil {
+		return nil, err
+	}
+	helpers := config.partialV2.CredentialHelpers
+	if len(helpers) == 0 {
+		helpers = append(helpers, AuthenticationFileHelper)
+	}
+	return helpers, nil
+}
+
+// FindCredentialHelpers returns a list of credential helpers for the
+// specified registry.  The helpers are intended to be contacted in the
+// returned order.
+//
+// Note that the global top-level credential helpers are ignored.
+func FindCredentialHelpers(sys *types.SystemContext, registry string) ([]string, error) {
```

Comment:
`CredentialHelpersForRegistry`? Not at all sure.

(Or should that be `…ForReference reference.Named`? Probably not, but think about the “multiple credentials for different tenants on `quay.io` problem. OTOH I don’t think we have a _solution_ for that either way, so maybe let’s add a new API only when we have one.)